### PR TITLE
fix: Fixes CFN autogen for creating new resources

### PR DESCRIPTION
**Pull request by maastha**
## Proposed changes

_Jira ticket:_ https://jira.mongodb.org/browse/CLOUDP-231641

This PR:
- fixes circular dependency errors in OpenAPI spec by using [transformed Atlas yaml API spec](https://raw.githubusercontent.com/mongodb/atlas-sdk-go/main/openapi/atlas-api-transformed.yaml) (also used in TF scaffolding and Atlas SDK)
- Updates existing template for mapping.json to specify a `contentType` to correctly generate request and response body object schemas
- updates `templateMakeFile` to add relevant debugging flags
- updates autogen to always use latest openAPI spec to generate the resource
- updates autogen README
- removes compare API spec logic and unwanted API spec files 

Link to any related issue(s): 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

## Manual QA performed:

- [ ] cfn invoke for each of CRUDL/cfn test
- [ ] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [ ] Published to AWS private registry
- [ ] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [ ] Deleted stack to ensure resources are deleted
- [ ] Created multiple resources in same stack
- [ ] Validated in Atlas UI
- [ ] Included screenshots

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [ ] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

### CODE REVIEW

File: cfn-resources/autogen/schema-gen/main.go [hunk 1]
```diff
@@ -251,30 +241,30 @@ func main() {
...
-		}
-		if diff == "" {
-			err = errors.New("no difference found in OpenAPI Spec")
-			return compare, err
-		}
-		return compare, err
-	}
-	return false, nil
-}
+// func readArgs() (compare bool, err error) {
```
Comment (marcosuma) [inline cfn-resources/autogen/schema-gen/main.go:244]: should we delete this?

**Comment by marcosuma** [comment]
This is great! Thanks @maastha for proactively fixing this 💯 

Follow-up question: how confident are we this will work for future resources? Have you actually used it for the stream connection or was this fixed later?

**Review by marcosuma — APPROVED** [review]

File: cfn-resources/autogen/README.md [hunk 2]
```diff
@@ -1,67 +1,68 @@
-# Auto Generate cfn template and generic GO Code
+# Using autogen
```
Comment (andreaangiolillo) [inline cfn-resources/autogen/README.md:1]: During the onboarding to CFN we [documented](https://wiki.corp.mongodb.com/display/MMS/CFN+Development+Guide#CFNDevelopmentGuide-AutomatedApproach) these limitations for the autogen script:

 ```
Limitations of the current implementations: [INTMDB-562](https://jira.mongodb.org/browse/INTMDB-562)

The schema-gen code has the following problem which would require changing the logic of the script:
Example: we have the property connectionStrings.privateEndpoint in [createCluster ](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Multi-Cloud-Clusters/operation/createCluster)(Response 201), which ref to ApiAtlasClusterDescriptionConnectionStringsPrivateEndpointView . Unfortunately, the definition of ApiAtlasClusterDescriptionConnectionStringsPrivateEndpointView is not added to the JSON schema file.
the code does not traverse a view property tree but it stops at the 2nd level

the autogen code adds additional boilercode to the code generated by cfn generate but it still requires several manual changes
```

I am wondering if this PR addresses these limitations as well

File: cfn-resources/autogen/schema-gen/main.go (hunk 1 above)
Comment (andreaangiolillo) [reply cfn-resources/autogen/schema-gen/main.go:244]: +1

File: cfn-resources/autogen/README.md [hunk 3]
```diff
@@ -1,67 +1,68 @@
...
 ```bash
-        make create dirName=custom-db-role resourceName=customdbrole typeName=CustomDBRole
+        make create dirName=stream-connection resourceName=streamconnection typeName=StreamConnection
 ```
 
-After executing the command 2 inputs are requested, 
-- Language (go is required)
-- Go Import Path
 
+After executing the command you need to provide the Go import path, which will be prompted on the terminal, for example, `github.com/mongodb/mongodbatlas-cloudformation-resources/stream-connection`
```
Comment (andreaangiolillo) [inline cfn-resources/autogen/README.md:53]: Could we also include the changes that we expect the developer still have to include in the generated code?

File: cfn-resources/autogen/README.md [hunk 4]
```diff
@@ -1,67 +1,68 @@
-# Auto Generate cfn template and generic GO Code
+# Using autogen 
 
-## How to generate Resource setup and Respective Go template for a new Resource
+## Autogenerating CFN resource schema and generic GO code for a new resource:
```
Comment (lantoli) [inline cfn-resources/autogen/README.md:3]:
```suggestion
## Autogenerating CFN resource schema and generic Go code for a new resource:
```

**Review by lantoli — APPROVED** [review]

File: cfn-resources/autogen/schema-gen/mapping.json [hunk 5]
```diff
@@ -1,260 +1,12 @@
...
-      "typeName": "CloudProviderAccess",
-      "openApiPath": [
-        "/api/atlas/v1.0/groups/{groupId}/cloudProviderAccess",
-        "/api/atlas/v1.0/groups/{groupId}/cloudProviderAccess/{cloudProvider}/{roleId}",
-        "/api/atlas/v1.0/groups/{groupId}/cloudProviderAccess/{roleId}"
-      ]
+        "/api/atlas/v2/groups/{groupId}/streams/{tenantName}/connections",
+        "/api/atlas/v2/groups/{groupId}/streams/{tenantName}/connections/{connectionName}"
+      ],
+      "contentType": "application/vnd.atlas.2023-02-01+json"
```
Comment (AgustinBettati) [inline cfn-resources/autogen/schema-gen/mapping.json:9]: like this approach

File: cfn-resources/go.mod [hunk 6]
```diff
@@ -10,9 +10,9 @@ require (
 	github.com/aws/aws-sdk-go-v2/service/cloudformation v1.47.0
```
Comment (AgustinBettati) [inline cfn-resources/go.mod:10]: Is it possible that this PR is missing the adjustments to the `templateMakeFile`? Would make sure to also include the `create-test-resources` `delete-test-resources` and `run-contract-testing` make commands as they are present for all resources that run with contract tests in CI.

File: cfn-resources/autogen/README.md (hunk 3 above)
Comment (maastha) [reply cfn-resources/autogen/README.md:53]: not sure what you mean here? We mention in the limitataions that the generated code would need to be manually adjusted already, is that what you mean?

File: cfn-resources/autogen/schema-gen/main.go (hunk 1 above)
Comment (maastha) [reply cfn-resources/autogen/schema-gen/main.go]: done

**Comment by maastha** [comment]
>Follow-up question: how confident are we this will work for future resources? Have you actually used it for the stream connection or was this fixed later?

While this was fixed later, I did generate schema for stream-connection and it was successful, of course, needless to say it required some manual adjustments as is called out in the [limitations](https://wiki.corp.mongodb.com/display/MMS/CFN+Development+Guide#CFNDevelopmentGuide-AutomatedApproach) Andrea mentioned, but the schema was generated correctly as per the API spec and request/response body object structures

File: cfn-resources/autogen/README.md (hunk 2 above)
Comment (maastha) [reply cfn-resources/autogen/README.md:1]: no this PR doesn't address that limitation

**Review by lantoli — DISMISSED** [review]

File: cfn-resources/autogen/README.md (hunk 3 above)
Comment (andreaangiolillo) [reply cfn-resources/autogen/README.md:53]: I am wondering what you think about adding the limitations here in the readme (if they are not already there)

Comment (maastha) [reply cfn-resources/autogen/README.md:53]: I don't have a strong opinion but I don't see too much value in adding here as well as they are captured in our internal wiki already but feel free to disagree

**Review by AgustinBettati — APPROVED** [review]

**Pull request merged**